PyP100/PyP100.py

```diff
@@ -0,0 +1,354 @@
+import requests
+from requests import Session
+
+from base64 import b64encode, b64decode
+import hashlib
+from Crypto.PublicKey import RSA
+import time
+import json
+from Crypto.Cipher import AES, PKCS1_OAEP, PKCS1_v1_5
+from . import tp_link_cipher
+import ast
+import pkgutil
+import uuid
+import json
+
+
+#Old Functions to get device list from tplinkcloud
+def getToken(email, password):
+	URL = "https://eu-wap.tplinkcloud.com"
+	Payload = {
+		"method": "login",
+		"params": {
+			"appType": "Tapo_Ios",
+			"cloudUserName": email,
+			"cloudPassword": password,
+			"terminalUUID": "0A950402-7224-46EB-A450-7362CDB902A2"
+		}
+	}
+
+	return requests.post(URL, json=Payload).json()['result']['token']
+
+def getDeviceList(email, password):
+	URL = "https://eu-wap.tplinkcloud.com?token=" + getToken(email, password)
+	Payload = {
+		"method": "getDeviceList",
+	}
+
+	return requests.post(URL, json=Payload).json()
+
+ERROR_CODES = {
+	"0": "Success",
+	"-1010": "Invalid Public Key Length",
+	"-1012": "Invalid terminalUUID",
+	"-1501": "Invalid Request or Credentials",
+	"1002": "Incorrect Request",
+	"-1003": "JSON formatting error "
+}
+
+class P100():
```

Comment:
Some of the functions in this class look like helpers. Maybe we can put them in a separate namespace? This would reduce the size of the class and make it easier to work with. This is just a suggestion. I may be wrong about the decomposition potential of this code. What do you think?